static/js/util/profile_edit_test.js

```diff
@@ -428,91 +414,240 @@ describe('Profile Editing utility functions', () => {
+        date: "28",
         month: "13",
         year: "2066"
       });
     });
 
+    it("uses a date of 1 if omitDate is true", () => {
+      that.props.profile.date_of_birth = "2066-02-28";
+      rerender(true);
+      monthTextField.props.onChange({target: {value: "13"}});
+      rerender(true);
+      monthTextField.props.onChange({target: {value: "1"}});
+
+      assert.deepEqual(that.props.profile.date_of_birth, "2066-01-01");
+      // date is 28 here but it's discarded in the timestamp above, resulting in 01
+      assert.deepEqual(that.props.profile.date_of_birth_edit, {
+        month: "1",
+        year: 2066,
+        date: 28
+      });
+    });
+
+    it("uses validateDate for validation", () => {
+      rerender(false);
+      dateTextField.props.onChange({target: {value: "a date"}});
+      assert(validateDateSpy.calledWith("a date"));
+    });
+
     it("uses validateMonth for validation", () => {
+      rerender(false);
       monthTextField.props.onChange({target: {value: "a month"}});
       assert(validateMonthSpy.calledWith("a month"));
     });
 
     it("uses validateYear for validation", () => {
+      rerender(false);
       yearTextField.props.onChange({target: {value: "a year"}});
       assert(validateYearSpy.calledWith("a year"));
     });
+
+    it("uses moment.js for validation", () => {
+      rerender(false);
+      that.props.profile.date_of_birth = "2066-02-28";
+      rerender(false);
+      dateTextField.props.onChange({target: {value: "31"}});
+      rerender(false);
+
+      // Feb 31 is not a valid date, and only moment.js knows that
+      assert.deepEqual(that.props.profile.date_of_birth, null);
+      assert.deepEqual(that.props.profile.date_of_birth_edit, {
+        date: "31",
+        month: 2,
+        year: 2066
+      });
+    });
+
+    it("treats an empty date string as deleting the format text but not edit data", () => {
+      that.props.profile.date_of_birth = "2066-02-28";
+      rerender(false);
+      dateTextField.props.onChange({target: {value: ""}});
+      assert.deepEqual(that.props.profile.date_of_birth, null);
+      assert.deepEqual(that.props.profile.date_of_birth_edit, {
+        date: "",
+        month: "2",
+        year: "2066"
+      });
+    });
+
+    it("treats an empty month string as deleting the format text but not edit data", () => {
+      that.props.profile.date_of_birth = "2066-02-28";
+      rerender(false);
+      monthTextField.props.onChange({target: {value: ""}});
+      assert.deepEqual(that.props.profile.date_of_birth, null);
+      assert.deepEqual(that.props.profile.date_of_birth_edit, {
+        date: "28",
+        month: "",
+        year: "2066"
+      });
+    });
+
+    it("treats an empty year string as deleting the format text but not edit data", () => {
+      that.props.profile.date_of_birth = "2066-02-28";
+      rerender(false);
+      yearTextField.props.onChange({target: {value: ""}});
```

Comment:
do we ever use helper methods in our test files? i kinda feel like we should. lots of repetition with this case and the two above. i think at the least it would be nice to have a helper method for the assertions which takes an object of expected values